File: ZooJava/src/test/java/com/example/zoojava/repositories/employees/EmployeesRepositoryImplTest.java
package com.example.zoojava.repositories.employees;

import com.example.zoojava.managers.DataBaseManager;
import com.example.zoojava.models.Employee;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.sql.SQLException;
import java.time.LocalDate;

import static org.junit.jupiter.api.Assertions.*;

class EmployeesRepositoryImplTest {
    private final EmployeesRepository repository = EmployeesRepositoryImpl.getInstance();
    private final Employee test = new Employee("Paco","paco.paquito@gmail.com","12345", LocalDate.now(),true);


    @AfterEach
    void tearDown() {
        try {
            repository.removeAll();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }
    @BeforeEach
    void setUp(){
        try {
            repository.removeAll();
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }
    }

    @Test
    void add() throws SQLException {
        var add = repository.add(test);
        assertAll(
                () -> assertEquals(add.getName(),test.getName()),
                () -> assertEquals(add.getEmail(),test.getEmail()),
                () -> assertEquals(add.getPassword(),test.getPassword()),
                () -> assertEquals(add.getBirthDate(),test.getBirthDate())
                );
    }


    @Test
    void remove() throws SQLException {
        var add = repository.add(test);
        var eliminate = repository.remove(test);
        assertAll(
                () -> assertEquals(eliminate.getName(),test.getName()),
                () -> assertEquals(eliminate.getEmail(),test.getEmail()),
                () -> assertEquals(eliminate.getPassword(),test.getPassword()),
                () -> assertEquals(eliminate.getBirthDate(),test.getBirthDate())
        );
    }

    @Test
    void findAll() throws SQLException {
        repository.add(test);
        var list = repository.findAll();
        assertAll(
                () -> assertEquals(list.size(),1),
                () -> assertEquals(list.get(0).getName(),test.getName()),
                () -> assertEquals(list.get(0).getEmail(),test.getEmail()),
                () -> assertEquals(list.get(0).getPassword(),test.getPassword()),
                () -> assertEquals(list.get(0).getBirthDate(),test.getBirthDate())
        );
    }

    @Test
    void findByEmail() throws SQLException {
        repository.add(test);
        var find = repository.findByEmail(test.getEmail());
        assertAll(
                () -> assertEquals(find.getName(),test.getName()),
                () -> assertEquals(find.getEmail(),test.getEmail()),
                () -> assertEquals(find.getPassword(),test.getPassword()),
                () -> assertEquals(find.getBirthDate(),test.getBirthDate())
        );
    }


    @Test
    void findByEmailNull() throws SQLException {
        var find = repository.findByEmail(test.getEmail());
        assertNull(find);
    }
}

File: ZooKotlin/src/test/kotlin/com/example/zookotlin/repositories/employees/EmployeesRepositoryImplTest.kt
package com.example.zookotlin.repositories.employees

import com.example.zookotlin.models.Employee
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.function.Executable
import java.time.LocalDate

internal class EmployeesRepositoryImplTest {
    private val repository: EmployeesRepository = EmployeesRepositoryImpl.getInstance()!!
    private val test: Employee = Employee("Paco", "paco.paquito@gmail.com", "12345", LocalDate.now(), true)


    @AfterEach
    fun tearDown() {
        repository.removeAll()
    }


    @BeforeEach
    fun setUp() {
        repository.removeAll()
    }

    @Test
    fun add() {
        val add = repository.add(test)
        assertAll(
            Executable { assertEquals(add!!.getName(), test.getName()) },
            Executable { assertEquals(add!!.getEmail(), test.getEmail()) },
            Executable { assertEquals(add!!.getPassword(), test.getPassword()) },
            Executable { assertEquals(add!!.getBirthDate(), test.getBirthDate()) }
        )
    }


    @Test
    fun remove() {
        val add = repository.add(test)
        val eliminate = repository.remove(test)
        assertAll(
            Executable { assertEquals(eliminate!!.getName(), test.getName()) },
            Executable { assertEquals(eliminate!!.getEmail(), test.getEmail()) },
            Executable { assertEquals(eliminate!!.getPassword(), test.getPassword()) },
            Executable { assertEquals(eliminate!!.getBirthDate(), test.getBirthDate()) }
        )
    }

    @Test
    fun findAll() {
        repository.add(test)
        val list = repository.findAll()
        assertAll(
            Executable { assertEquals(1, list.size) },
            Executable { assertEquals(list[0].getName(), test.getName()) },
            Executable { assertEquals(list[0].getEmail(), test.getEmail()) },
            Executable { assertEquals(list[0].getPassword(), test.getPassword()) },
            Executable { assertEquals(list[0].getBirthDate(), test.getBirthDate()) }
        )
    }

    @Test
    fun findByEmail() {
        repository.add(test)
        val find = repository.findByEmail(test.getEmail())
        assertAll(
            Executable { assertEquals(find!!.getName(), test.getName()) },
            Executable { assertEquals(find!!.getEmail(), test.getEmail()) },
            Executable { assertEquals(find!!.getPassword(), test.getPassword()) },
            Executable { assertEquals(find!!.getBirthDate(), test.getBirthDate()) }
        )
    }


    @Test
    fun findByEmailNull() {
        val find = repository.findByEmail(test.getEmail())
        assertNull(find)
    }
}